realm/realm-library/src/androidTest/java/io/realm/RealmIntegerTests.java

```diff
@@ -0,0 +1,228 @@
+public class RealmIntegerTests {
+
+    @Rule
+    public final TestRealmConfigurationFactory configFactory = new TestRealmConfigurationFactory();
+    @Rule
+    public final RunInLooperThread looperThread = new RunInLooperThread();
+
+    private Realm realm;
+
+    @Before
+    public void setUp() throws Exception {
+        RealmConfiguration realmConfig = configFactory.createConfiguration();
+        realm = Realm.getInstance(realmConfig);
+    }
+
+    @After
+    public void tearDown() throws Exception {
+        if (realm != null) {
+            if (realm.isInTransaction()) {
+                realm.cancelTransaction();
+            }
+            realm.close();
+        }
+    }
+
+    @Test
+    public void basic() {
+        RealmInteger ri1 = new RealmInteger(10);
+        RealmInteger ri2 = new RealmInteger("10");
+        assertEquals(ri1, ri2);
+
+        ri1.set(15);
+        ri1.decrement(2);
+        ri2.increment(3);
+        assertEquals(ri1, ri2);
+    }
+
+    @Test
+    public void getters() {
+        RealmInteger ri = new RealmInteger(0x04433332211L);
+
+        // positive
+        assertEquals(ri.longValue(), 0x04433332211L);
+        assertEquals(ri.intValue(), 0x033332211);
+        assertEquals(ri.shortValue(), 0x02211);
+        assertEquals(ri.byteValue(), 0x011);
+
+        assertEquals(ri.floatValue(), 2.92916756E11F);
+        assertEquals(ri.doubleValue(), 2.92916765201E11);
+
+        // negative
+        ri.set(0x8888444483338281L);
+        assertEquals(ri.longValue(), 0x8888444483338281L);
+        assertEquals(ri.intValue(), 0x083338281);
+        assertEquals(ri.shortValue(), (short) 0xf8281);
+        assertEquals(ri.byteValue(), (byte) 0xf81);
+
+        assertEquals(ri.floatValue(), -8.6085554E18F);
+        assertEquals(ri.doubleValue(), -8.6085556266690468E18);
+    }
+
+    @Test
+    public void compareTo() {
+        RealmInteger ri1 = new RealmInteger(10);
+        RealmInteger ri2 = new RealmInteger("11");
+
+        assertEquals(ri1.compareTo(ri2), -1);
+        ri2.decrement(1);
+        assertEquals(ri1.compareTo(ri2), 0);
+        ri2.decrement(1);
+        assertEquals(ri1.compareTo(ri2), 1);
+    }
+
+    @Ignore("not yet implemented")
+    @Test
+    public void importedInteger() {
+        final RealmInteger ri = new RealmInteger(5);
+
+        realm.executeTransaction(
+                new Realm.Transaction() {
+                    @Override
+                    public void execute(Realm realm) {
+                        realm.createObject(AllTypes.class).setColumnRealmInteger(ri);
+                    }
+                });
+
+        realm.executeTransaction(
+                new Realm.Transaction() {
+                    @Override
+                    public void execute(Realm realm) {
+                        realm.where(AllTypes.class).findFirst().getColumnRealmInteger().decrement(1);
+                    }
+                });
+
+        assertEquals(ri.longValue(), 4);
```

Comment:
Yes, this should be the resulting value. Any reason you do this in 2 transactions?